src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -159,6 +159,28 @@ public void MulticastInterface_Set_InvalidIndex_Throws()
             }
         }
 
+        [OuterLoop] // TODO: Issue #11345
+        [Fact]
+        public void MulticastInterface_Set_WithIPv6()
+        {
+            using (Socket s = new Socket(AddressFamily.InterNetworkV6, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(1));
```

Comment:
`IPAddress.HostToNetworkOrder` should not be needed here:

For IPV6_MULTICAST_IF:
> The input value for setting this option is a 4-byte interface index of the desired outgoing interface in **host byte order**.

Whereas for IP_MULTICAST_IF:
> The input value for setting this option is a 4-byte IPv4 address in network byte order. This DWORD parameter can also be an interface index in **network byte order**.

Reply:
Push a fix for that